File: mslibrary-kotlindocs/src/main/java/kr/msleague/message/adapter/CraftPlayerAdapter.java
package kr.msleague.message.adapter;

import kr.msleague.message.api.MSMessageAdapter;
import org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer;

/**
 * MSLibrary 에서 기본적으로 제공하는 CraftPlayer 어댑터입니다.
 */
public class CraftPlayerAdapter implements MSMessageAdapter<CraftPlayer> {
    @Override
    public String reformat(String origin, CraftPlayer obj) {
        return "";
    }
}

File: mslibrary-message/src/main/java/kr/msleague/message/adapter/CraftPlayerAdapter.kt
package kr.msleague.message.adapter

import kr.msleague.message.api.MSMessageAdapter
import kr.msleague.util.extensions.roundString
import org.bukkit.craftbukkit.v1_12_R1.entity.CraftPlayer

class CraftPlayerAdapter : MSMessageAdapter<CraftPlayer> {

    override fun request(obj: CraftPlayer, placeHolder: String): String? {
        return when (placeHolder) {
            "player_name" -> obj.name
            "player_health" -> obj.health.roundString()
            "player_max_health" -> obj.health.roundString()
            else -> null
        }
    }
}